Db2dev

dwadler opened the pull request:
## Description
Fix DB2 provider crash per issue #20337.  Migrate changed logic from mssql provider to ensure database connections are in distinct threads and query logic in the same thread.

## Checklist

- [x ] Commit messages are descriptive and explain the rationale for changes
- [x ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [na ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [na ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ na] New unit tests have been added for core changes
- [x ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

dwadler committed:
DB2 fixes for #20337

dwadler committed:
run prepare-commit on DB2 fixes for #20337

nyalldawson commented:
@dwadler there were a couple of related follow-up fixes which should also be ported:

https://github.com/qgis/QGIS/commit/a5ff6db24066642e2b5952b22bfe069013ec9a0b#diff-b9a20555ff03b8ffd1bf952cf6b98374

https://github.com/qgis/QGIS/commit/c782f2782016f9be0f127de25a42e48e82ead378#diff-b9a20555ff03b8ffd1bf952cf6b98374

dwadler commented:
Is there a straightforward way to find out who change what and when with 
particular source? I'm appreciative of whoever updated the DB2 provider 
between QGIS 2.x and 3.x but the code is considerably different from 
what I originally implemented.

I'm hoping that GIT can be my friend rather than the enemy. :-)


On 11/11/2018 2:16 PM, Nyall Dawson wrote:
>
> @dwadler <https://github.com/dwadler> there were a couple of related 
> follow-up fixes which should also be ported:
>
> a5ff6db#diff-b9a20555ff03b8ffd1bf952cf6b98374 
> <https://github.com/qgis/QGIS/commit/a5ff6db24066642e2b5952b22bfe069013ec9a0b#diff-b9a20555ff03b8ffd1bf952cf6b98374>
>
> c782f27#diff-b9a20555ff03b8ffd1bf952cf6b98374 
> <https://github.com/qgis/QGIS/commit/c782f2782016f9be0f127de25a42e48e82ead378#diff-b9a20555ff03b8ffd1bf952cf6b98374>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/qgis/QGIS/pull/8459#issuecomment-437696524>, or 
> mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AHAx6IxNTZqDeEXCwDJvN_5kISa2M3_Wks5uuHejgaJpZM4YYjV_>.
>



---
This email has been checked for viruses by Avast antivirus software.
https://www.avast.com/antivirus

rouault commented:
- Browse through commits in the provider:
https://github.com/qgis/QGIS/commits/master/src/providers/db2
(or ```git log src/providers/db2```)

- Identify who modified last each line on a file. Use blame. e.g: https://github.com/qgis/QGIS/blob/master/src/providers/db2/qgsdb2provider.cpp
Or  ```git blame src/providers/db2/qgsdb2provider.cpp```

nyalldawson commented:
@dwadler 

I don't believe there's been any db2 specific fixes/improvements since your initial work. The only changes which have been made to the provider have been as a result of the work done in QGIS 3.0 -- adapting the provider for Qt 5, c++11, and the QGIS API changes.

dwadler committed:
Update src/providers/db2/qgsdb2provider.h

dwadler committed:
Update src/providers/db2/qgsdb2provider.cpp

dwadler committed:
Fix indentation - issue #20337

Hopefully this is correct.
I'm not having much luck getting prepare-commit.sh or astyle.sh to run on Windows.
